lib/index.js

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright (c) 2017 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bedrock = require('bedrock');
+const brLedger = require('bedrock-ledger');
+const equihashSigs = require('equihash-signature');
+const validate = require('bedrock-validation').validate;
+const BedrockError = bedrock.util.BedrockError;
+
+require('./config');
+
+bedrock.events.on('bedrock.start', () => brLedger.use('EquihashValidator2017', {
+  type: 'validator',
+  api: api
+}));
+
+const api = {};
+// NOTE: only exported for tests
+module.exports = api;
+
+api.mustValidateEvent = (event, validatorConfig, options, callback) => {
+  if(typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+  api.validateConfiguration(validatorConfig, err => {
+    if(err) {
+      return callback(err);
+    }
+    if(!validatorConfig.eventFilter.some(f =>
+      f.type === 'EventTypeFilter' && f.eventType.includes(event.type))) {
+      return callback(null, false);
+    }
+    callback(null, true);
+  });
+};
+
+api.validateConfiguration = (validatorConfig, callback) =>
+  validate('ledger-validator-equihash-config', validatorConfig, callback);
+
+api.validateEvent = (event, validatorConfig, callback) => {
+  if(!validatorConfig.eventFilter.some(f =>
```

Comment:
I agree -- it is the validator's responsibility to that an event must be validated (or not). But that means, to me, that we don't need an extra check in the actual `validateEvent` call here that was already done in `mustValidateEvent`. And we don't want devs to have the extra cognitive overhead of having to check things twice.